rs/nns/governance/src/proposals/mod.rs

```diff
@@ -54,3 +59,82 @@ pub(crate) fn invalid_proposal_error(reason: &str) -> GovernanceError {
         format!("Proposal invalid because of {}", reason),
     )
 }
+
+/// Weighted voting power is just voting power * the proposal's (topic's) weight.
+///
+/// For example, suppose this returns ({42 => 3.14}, 99.9). This means that
+/// neuron 42 should get 3.14/99.9 of today's reward purse.
+///
+/// Non-essential fact: It is highly unusual for result.1 to be the sum of the
+/// values in result.0. The main reason they would not be equal is that not all
+/// neurons vote. Another (less imporant reason) is that some neurons lose
+/// voting power due to inactivity.
+pub fn sum_weighted_voting_power<'a>(
+    proposals: impl Iterator<Item = (ProposalId, Option<&'a ProposalData>)>,
+) -> (HashMap<NeuronId, /* exercised */ f64>, /* total */ f64) {
```

Comment:
IMHO, if you are adding comments about what these values mean, this implies it's time to make a struct that would carry that information properly. Maybe `WeightedVotingPower` would be a suitable name?

```suggestion
) -> (HashMap<NeuronId, WeightedVotingPower>) {
```

Reply:
I now realize you already approved -> I assume this means you leave it up to me. To me, this is not so complicated that it justifies a new struct. I think the mental weight of the indirection created by a new struct is too much for this.